### PR TITLE
[ACTION] Update Threads components to use newest API #4350

### PR DESCRIPTION
**Pull request by luancazarine**
Refactored:
  - App
  - Action Post Thread
  - Action Delete Thread

Created:
  - Action Post Chat Message

<a href="https://gitpod.io/#https://github.com/PipedreamHQ/pipedream/pull/4367"></a>

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/threads/actions/post-chat-message/post-chat-message.mjs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      chatID,
+      body,
+    } = this;
+    const chatMessage = await this.threads.postChatMessage({
+      $,
+      chatID,
+      body,
+    });
+
+    console.log("chatMessage: ", chatMessage);
```
Comment (vellames) [inline components/threads/actions/post-chat-message/post-chat-message.mjs:38]: Do you want this here?

**Review by vellames — CHANGES_REQUESTED** [review]
Hey! Looks like you forgot a console.log here. Not sure, but I am requesting changes just in case.

**Review by vellames — APPROVED** [review]
lgtm

**Comment by vunguyenhung** [comment]
Hi everyone, all test cases are passed! Ready for release! 

Test report
[Threads_4350.pdf](https://github.com/PipedreamHQ/pipedream/files/9698319/Threads_4350.pdf)

**Review by feyzullah — APPROVED** [review]

**Comment by dannyroosevelt** [comment]
I'm running into problems getting any of these components to work.

**Post Thread**
Screenshot by Dropbox Capture
And both prop descriptions could be improved. I can provide a recommendation.

**Post Chat Message**
Screenshot by Dropbox Capture
Screenshot by Dropbox Capture
Both prop descriptions could be improved.

**Delete Thread**
Screenshot by Dropbox Capture
Screenshot by Dropbox Capture
The prop description is outdated I believe (it references threads.com, but it should reference trythreads.com)

**Comment by vunguyenhung** [comment]
Sorry @dannyroosevelt, I have created a new PR to fix the Post Thread action, as well as improve the description to `trythreads.com` https://github.com/PipedreamHQ/pipedream/pull/4413

File: components/threads/threads.app.mjs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      type: "string",
+      label: "Thread ID",
+      description:
+        "Navigate to your thread on the Threads website. The URL will be threads.com/${thread_id}",
+    },
+    chatID: {
+      type: "string",
+      label: "Chat ID",
+      description:
+        "This is ID of the chat you want to post to. An easy way to find a chatID for a chat is to navigate to that chat on the Threads website. The URL will be trythreads.com/messages/<your_chat_id>",
```
Comment (dannyroosevelt) [inline components/threads/threads.app.mjs:43]: >"To find the chat ID, open the Threads chat in your browser and copy the ID: `https://trythreads.com/messages/{your_chat_id}`"

File: components/threads/actions/post-thread/post-thread.mjs [hunk 3]
```diff
@@ -1,23 +1,17 @@
-const threads = require("../../threads.app.js");
+import threads from "../../threads.app.mjs";
 
-module.exports = {
+export default {
   key: "threads-post-thread",
   name: "Post a Thread",
   description: "Post a new thread to a specific forum",
```
Comment (dannyroosevelt) [inline components/threads/actions/post-thread/post-thread.mjs:6]: >"Post a new thread to a specific channel",

File: components/threads/actions/post-chat-message/post-chat-message.mjs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      ],
+    },
+    body: {
+      propDefinition: [
+        threads,
+        "body",
+      ],
+      label: "Message",
+      description:
+        "This is the body of the message. To format the body, you can send markdown.",
```
Comment (dannyroosevelt) [inline components/threads/actions/post-chat-message/post-chat-message.mjs:24]: Compose your chat message here, and include markdown if you'd like. See [here](https://github.com/ThreadsHQ/api-documentation#2-1) for examples.

**Review by dannyroosevelt — COMMENTED** [review]
I added comments on suggested changes to the component and prop descriptions

**Pull request merged**